integration/consul_catalog_test.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	checker "github.com/vdemeester/shakers"
+	check "gopkg.in/check.v1"
+)
+
+// Consul catalog test suites
+type ConsulCatalogSuite struct {
+	BaseSuite
+	client *api.Client
+}
+
+func (s *ConsulCatalogSuite) SetUpSuite(c *check.C) {
+	s.createComposeProject(c, "consul_catalog")
+	s.composeProject.Up()
+
+	config := api.DefaultConfig()
+	config.Address = "127.0.0.1:8500"
```

Comment:
Yeah exactly. I'm working on a Go lib' to simplify that (and we will integrate it into traefik when the times come) but yeah that's basically the gist of it, listing container filtering (by the labels compose put I would use) and get informations about it. :stuck_out_tongue_closed_eyes: